core/src/main/java/com/graphhopper/routing/util/OSMParsers.java

```diff
@@ -87,6 +88,17 @@ else if ("platform".equals(way.getTag("railway")))
             return false;
     }
 
+    public Predicate<ReaderRelation> getRelationFilterForWays() {
+        if (ignoredHighways.contains("pedestrian") && ignoredHighways.contains("footway"))
+            // if are both ignored we can skip pass0 altogether
+            return null;
```

Comment:
The 0th pass is skipped entirely if `pedestrian` and `footway` are both added to the list of ignored highways, because >80% of these highway relations use one of these two highway values. Then again, there are also around [10k relations tagged as `service,tertiary,platform,bus_stop...`](https://taginfo.openstreetmap.org/keys/highway?filter=relations#values). I'm not sure if we really need them for motorized-only routing (i.e. when pedestrian+footway are excluded). In case they are not excluded we would accept all highway relations anyway with this PR.